### PR TITLE
Improve CI

### LINKED ISSUE #159: Improve CI stability
We often get the following error message for our CI builds

[!] Operation not permitted - /Users/travis/Library/Application Support/CallHistoryTransactions (Errno::EPERM)

manually restarting the build often resolves the issue but it's something we should investigate and fix

### PR DESCRIPTION
**Pull request by fbernutz**
Fixes #159 

### Fix unreliable CI 
Use a workaround to resolve SPM Dependencies. This is a timing issue in fastlane and the workaround should be removed when the issue is fixed by fastlane. 

| Before | After |
|---|---|
|Screen Shot 2019-10-19 at 10 41 01|Screen Shot 2019-10-19 at 10 57 26|

### Improve and update CI
- Update Fastlane 
- Run `bundle install` instead of `bundle update` in CI
- Install Dependencies on `install` 
(see here for more information: https://docs.travis-ci.com/user/job-lifecycle/#customizing-the-installation-phase) 
- Use `Build For Testing` Configuration
This speeds up running the unit tests with fastlane from ~170s to ~70s (in CI) 🚀 
- Cache Bundler in CI
(see here for more information: https://docs.travis-ci.com/user/caching/#enabling-bundler-caching)

### Results
🚀These improvements speeds up the CI:
- the job for _unit tests_ from ~4min 10sec to ~2min 15sec
- the job for _UI tests_ from ~5min 5sec to ~4min 10sec

### Before merging the PR

- [ ] Did you update the [CHANGELOG.md](https://github.com/criticalmaps/criticalmaps-ios/blob/master/CHANGELOG.md)?

(I don't think this is relevant for the changelog, right?)

### CODE REVIEW

**Comment by lennet** [comment]
Thank you for working on it! There seems to be an update for fastlane. Maybe that fixes the issue

**Comment by lennet** [comment]
The problem might be related to The problem might be related to https://github.com/fastlane/fastlane/issues/15301 . There are also some possible workarounds described in the issue

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,24 @@
 language: objective-c 
-osx_image: xcode11
+osx_image: xcode11.2
 
 before_install:
-  - bundle update
+  - bundle install
```
Comment (mltbnz) [inline .travis.yml]: good idea

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,24 @@
 language: objective-c 
-osx_image: xcode11
+osx_image: xcode11.2
```
Comment (mltbnz) [inline .travis.yml]: isn't that still beta?

Comment (fbernutz) [reply .travis.yml]: oh true, thanks for the hint! I got tricked by the travis documentation where it is the latest xcode version: https://docs.travis-ci.com/user/reference/osx/#xcode-version

**Comment by fbernutz** [comment]
Screen Shot 2019-10-19 at 10 04 56
Maybe this could be a relevant setting in travis for this project. This will cancel all pending jobs when a new commit is pushed. As I'm not able to cancel jobs in this repo, this can be annoying for contributors if you have to wait for old commits finishing their jobs. Well, this is mostly annoying when you test the CI (and of course it's my fault to run these many jobs on each commit).

**Comment by lennet** [comment]
> Screen Shot 2019-10-19 at 10 04 56
> 
> Maybe this could be a relevant setting in travis for this project. This will cancel all pending jobs when a new commit is pushed. As I'm not able to cancel jobs in this repo, this can be annoying for contributors if you have to wait for old commits finishing their jobs. Well, this is mostly annoying when you test the CI (and of course it's my fault to run these many jobs on each commit).

Both settings are already enabled

**Comment by fbernutz** [comment]
> > Screen Shot 2019-10-19 at 10 04 56
> > Maybe this could be a relevant setting in travis for this project. This will cancel all pending jobs when a new commit is pushed. As I'm not able to cancel jobs in this repo, this can be annoying for contributors if you have to wait for old commits finishing their jobs. Well, this is mostly annoying when you test the CI (and of course it's my fault to run these many jobs on each commit).
> 
> Both settings are already enabled

Mh interesting, maybe this travis feature just doesn't work like I want it to work. So nevermind 👍 Btw there is an open issue to cancel jobs which are already running (https://github.com/travis-ci/travis-ci/issues/8712).

**Pull request merged**